components/raftstore/src/store/peer.rs

```diff
@@ -1794,7 +1827,6 @@ where
         // needs to be sent to the apply system.
         // Always sending snapshot task behind apply task, so it gets latest
         // snapshot.
-        // TODO: maybe we should move this code to other place to make the logic more clear.
```

Comment:
This TODO work has not been done yet. BTW, I find a bug that the snapshot may not be generated if there is not new committed entry. So the code below should be moved outside. Furthermore, it is unnecessary to send the snapshot task to apply fsm. I will fix them later.

Reply:
I revert this change, PTAL again